dynamiq/storages/vector/pgvector/filters.py

```diff
@@ -0,0 +1,398 @@
+    """
+    Parses metadata fields by using the ->> operator to access keys in the metadata JSONB field
+    and casting them to the correct type.
+
+    Args:
+        field (str): The field name.
+        value (Any): The value to determine the type for casting.
+
+    Returns:
+        str: The modified field with the correct type casting.
+    """
+
+    # Get the field name from the metadata field
+    field_name = field.split(".", 1)[-1]
+
+    # Retrieve the field using the ->> operator
+    field = f"metadata->>'{field_name}'"
+
+    # Determine the type for casting
+    type_value = None
+    if isinstance(value, list) and value:
+        type_value = PYTHON_TO_PGVECTOR_DATA_TYPE.get(type(value[0]))
+    else:
+        type_value = PYTHON_TO_PGVECTOR_DATA_TYPE.get(type(value))
+
+    # Cast the field to the correct type
+    if type_value:
+        field = f"({field})::{type_value}"
+
+    return field
+
+
+def _validate_equality_value(value: Any) -> None:
+    """
+    Validate value type for equality operators.
+
+    Args:
+        value (Any): Value to be validated.
+
+    Raises:
+        VectorStoreFilterException: If the value type is not supported.
+    """
+    supported_types = (str, int, float, bool)
+    if not isinstance(value, supported_types):
+        msg = f"Unsupported type for comparison: {type(value)}. " f"Types supported by PGVector are: {supported_types}"
+        raise VectorStoreFilterException(msg)
+
+
+def _validate_list_value(field: str, value: Any, operator: str) -> None:
+    """
+    Validate list values for 'in' and 'not in' operators.
+
+    Args:
+        field (str): Field name.
+        value (Any): Value to be validated.
+        operator (str): Comparison operator.
+
+    Raises:
+        VectorStoreFilterException: If the value is not a list or contains unsupported types.
+    """
+    if not isinstance(value, list):
+        msg = f"{field}'s value must be a list when using '{operator}' comparator in PGVector"
+        raise VectorStoreFilterException(msg)
+
+    supported_types = (int, float, str)
+    for v in value:
+        if not isinstance(v, supported_types):
+            msg = (
+                f"Unsupported type for '{operator}' comparison: {type(v)}. "
+                f"Types supported by PGVector are: {supported_types}"
+            )
+            raise VectorStoreFilterException(msg)
+
+
+def _validate_comparison_value(value: Any) -> None:
+    """
+    Common validation for comparison operators.
+
+    Args:
+        value (Any): Value to be validated.
+
+    Raises:
+        VectorStoreFilterException: If the value is of an unsupported type.
+    """
+    if isinstance(value, str):
+        try:
+            datetime.fromisoformat(value)
+        except (ValueError, TypeError) as exc:
+            msg = "Can't compare strings using operators '>', '>=', '<', '<='. "
+            raise VectorStoreFilterException(msg) from exc
+    if type(value) in [list, Jsonb]:
```

Comment:
Is it possible to use `isinstance(value, (list, Jsonb))` too?

Reply:
Sure, fixed that.